src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1283,7 +1283,31 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     LOG4CXX_ERROR(logger_, "Service URLs are empty! NOT sending PT to mobile!");
     return;
   }
-  SendMessageToSDK(pt_string, urls.front().url.front());
+
+  static size_t current_app = 0;
```

Comment:
@tkireva it's really hard to understand this code. It's unreadable. I'm not insist hardly but it would be better to rework this loop to decrease conditions complexity